### DISCUSSION TITLE
Play rtttl tones on buzzer

### DISCUSSION THREAD

**Discussion by irqnet** [arendst/Tasmota#16865] [Ideas]
Related to the documentation the buzzer command currently only supports "playing" beeps as it is also foreseen to be used for LED blink notifications.

Would it be possible (like in ESPHome - https://esphome.io/components/rtttl.html) to implement playing rtttl ringtones with Tasmota?

**Comment by sfromis** [comment]
Normally, "possible?" is not a useful question, as the issue is who has the interest and inclination to work on an implementation. Major parts of the open source project is the result of people wanting something, and creating a feature.

**Comment by Jason2866** [comment]
Probably not easy (impossible?) with Tasmota since fast changes of PWM needed.

**Comment by BenJamesAndo** [comment]
Looks like it did get added https://github.com/arendst/Tasmota/pull/18423
However you need to compile Tasmota yourself to include [I2S Audio](https://tasmota.github.io/docs/I2S-Audio/).

**Reply by sfromis** [reply]
Except that I2S audio is not via the buzzer support.

However here's a RTTTL driver workable with a PWM "buzzer" on ESP32, coded in Berry:
https://github.com/arendst/Tasmota/pull/21375#issuecomment-2096458422

**Reply by BenJamesAndo** [reply]
Ah I see. Thanks for letting me know. I've tried to run the Berry script and it says it's played correctly, but I can't hear anything. I have a piezo buzzer on GPIO4. Do I simply change `gpio.pin_mode(15,gpio.OUTPUT)` to `gpio.pin_mode(4,gpio.OUTPUT)` ?
Am I supposed to set it up as a PWM under Configure Template?

**Reply by barbudor** [reply]
Yes, it looks like it is hard code to gpio 15
So change any gpio référence from 15 to 4

Berry doesn't need to gpio to be configured in Tasmota template

**Reply by sfromis** [reply]
If you prefer, you could also avoid the hardcoding of the pin, by configuring it as PWM 6 (avoid 1-5 as they make Tasmota see it as a "light"), and replacing the pin number by `gpio.pin(gpio.PWM1, 5)`

FTR, the Berry program is also among the examples Staars has on Github:
https://github.com/Staars/berry-examples/blob/main/gpio/ulanzi_rtttl.be

**Reply by Staars** [reply]
> If you prefer, you could also avoid the hardcoding of the pin, by configuring it as PWM 6 (avoid 1-5 as they make Tasmota see it as a "light"), and replacing the pin number by gpio.pin(gpio.PWM1, 5)

This is a good idea.
